### PR TITLE
Switch pkg/intel to fiano's version

### PR DESCRIPTION
**Pull request by xaionaro**
# Test Plan

## unit-tests
```
xaionaro@void:~/go/src/github.com/9elements/converged-security-suite$ go test ./... -count=1
?   	github.com/9elements/converged-security-suite/v2/cmd/cbnt-prov	[no test files]
?   	github.com/9elements/converged-security-suite/v2/cmd/pcr0tool	[no test files]
?   	github.com/9elements/converged-security-suite/v2/cmd/pcr0tool/commands	[no test files]
?   	github.com/9elements/converged-security-suite/v2/cmd/pcr0tool/commands/diff	[no test files]
?   	github.com/9elements/converged-security-suite/v2/cmd/pcr0tool/commands/diff/format	[no test files]
?   	github.com/9elements/converged-security-suite/v2/cmd/pcr0tool/commands/dumpfit	[no test files]
?   	github.com/9elements/converged-security-suite/v2/cmd/pcr0tool/commands/dumpregisters	[no test files]
?   	github.com/9elements/converged-security-suite/v2/cmd/pcr0tool/commands/dumpregisters/helpers	[no test files]
?   	github.com/9elements/converged-security-suite/v2/cmd/pcr0tool/commands/printnodes	[no test files]
?   	github.com/9elements/converged-security-suite/v2/cmd/pcr0tool/commands/sum	[no test files]
?   	github.com/9elements/converged-security-suite/v2/cmd/txt-prov	[no test files]
?   	github.com/9elements/converged-security-suite/v2/cmd/txt-suite	[no test files]
?   	github.com/9elements/converged-security-suite/v2/pkg/check	[no test files]
ok  	github.com/9elements/converged-security-suite/v2/pkg/diff	0.027s
ok  	github.com/9elements/converged-security-suite/v2/pkg/errors	0.060s
?   	github.com/9elements/converged-security-suite/v2/pkg/hwapi	[no test files]
?   	github.com/9elements/converged-security-suite/v2/pkg/log	[no test files]
?   	github.com/9elements/converged-security-suite/v2/pkg/mathtools	[no test files]
?   	github.com/9elements/converged-security-suite/v2/pkg/ostools	[no test files]
?   	github.com/9elements/converged-security-suite/v2/pkg/pcd	[no test files]
?   	github.com/9elements/converged-security-suite/v2/pkg/pcd/consts	[no test files]
?   	github.com/9elements/converged-security-suite/v2/pkg/pcd/helpers	[no test files]
ok  	github.com/9elements/converged-security-suite/v2/pkg/pcr	0.015s
?   	github.com/9elements/converged-security-suite/v2/pkg/pcr/types	[no test files]
ok  	github.com/9elements/converged-security-suite/v2/pkg/provisioning/cbnt	0.044s
?   	github.com/9elements/converged-security-suite/v2/pkg/provisioning/txt	[no test files]
ok  	github.com/9elements/converged-security-suite/v2/pkg/registers	1.354s
ok  	github.com/9elements/converged-security-suite/v2/pkg/test	0.017s
ok  	github.com/9elements/converged-security-suite/v2/pkg/tools	0.101s
ok  	github.com/9elements/converged-security-suite/v2/pkg/tpmdetection	0.025s
?   	github.com/9elements/converged-security-suite/v2/pkg/tpmeventlog	[no test files]
ok  	github.com/9elements/converged-security-suite/v2/pkg/uefi	0.009s [no tests to run]
?   	github.com/9elements/converged-security-suite/v2/pkg/uefi/consts	[no test files]
?   	github.com/9elements/converged-security-suite/v2/pkg/uefi/ffs	[no test files]
?   	github.com/9elements/converged-security-suite/v2/pkg/uefi/ffs/consts	[no test files]
```

## pcr0tool
```
xaionaro@void:~/go/src/github.com/9elements/converged-security-suite$ go run ./cmd/pcr0tool/ sum ~/firmware/F09_3A10.bin > /tmp/2.txt
xaionaro@void:~/go/src/github.com/9elements/converged-security-suite$ git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
xaionaro@void:~/go/src/github.com/9elements/converged-security-suite$ go run ./cmd/pcr0tool/ sum ~/firmware/F09_3A10.bin > /tmp/1.txt
xaionaro@void:~/go/src/github.com/9elements/converged-security-suite$ diff /tmp/[12].txt
```

## cbnt-prov

```
xaionaro@void:~/go/src/github.com/9elements/converged-security-suite$ go run ./cmd/cbnt-prov/ show-all ~/firmware/Y3DLP08.bin > /tmp/1.txt
xaionaro@void:~/go/src/github.com/9elements/converged-security-suite$ git checkout migrate/intel_to_fiano
Switched to branch 'migrate/intel_to_fiano'
Your branch is up to date with 'origin/migrate/intel_to_fiano'.
xaionaro@void:~/go/src/github.com/9elements/converged-security-suite$ go run ./cmd/cbnt-prov/ show-all ~/firmware/Y3DLP08.bin > /tmp/2.txt
xaionaro@void:~/go/src/github.com/9elements/converged-security-suite$ diff /tmp/[12].txt
1,33c1,30
< Firmware Interface Table
< ------------------------
< Type                      | Address              | Size     | Version    | Checksum valid  | Checksum
< -----------------------------------------------------------------------------------------------------
< FIT_header_entry          | 0x2020205f5449465f   | 28       | {"maj":1}  | false           | 241
< microcode_update_entry    | 0xffef8180           | 0        | {"maj":1}  | false           | 0
< microcode_update_entry    | 0xffefe580           | 0        | {"maj":1}  | false           | 0
< microcode_update_entry    | 0xfff04d80           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< skip_entry                | 0xffffffff           | 0        | {"maj":1}  | false           | 0
< startup_ACM_entry         | 0xffbc0000           | 0        | {"maj":1}  | false           | 0
< BIOS_startup_module_entry | 0xfff0b760           | 2        | {"maj":1}  | false           | 0
< BIOS_startup_module_entry | 0xffc00000           | 262144   | {"maj":1}  | false           | 0
< BIOS_policy_record        | 0x0                  | 0        | {"maj":1}  | false           | 0
< TXT_policy_record         | 0x2a050100710070     | 0        | {"maj":0}  | false           | 0
< key_manifest_record       | 0xfff0bd80           | 597      | {"maj":1}  | false           | 0
< boot_policy_manifest      | 0xfff0b800           | 941      | {"maj":1}  | false           | 0
<
---
> #   | Type                             | Address              | Size     | Version | Checksum valid  | Checksum
> ---------------------------------------------------------------------------------------------------------------
> 0   | FITHeaderEntry            (0x00) | 0x2020205f5449465f   | 28       | 0x0100  | false           | 241
> 1   | MicrocodeUpdateEntry      (0x01) | 0xffef8180           | 0        | 0x0100  | false           | 0
> 2   | MicrocodeUpdateEntry      (0x01) | 0xffefe580           | 0        | 0x0100  | false           | 0
> 3   | MicrocodeUpdateEntry      (0x01) | 0xfff04d80           | 0        | 0x0100  | false           | 0
> 4   | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 5   | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 6   | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 7   | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 8   | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 9   | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 10  | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 11  | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 12  | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 13  | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 14  | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 15  | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 16  | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 17  | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 18  | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 19  | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 20  | Skip                      (0x7F) | 0xffffffff           | 0        | 0x0100  | false           | 0
> 21  | SACM                      (0x02) | 0xffbc0000           | 0        | 0x0100  | false           | 0
> 22  | BIOSStartupModuleEntry    (0x07) | 0xfff0b760           | 2        | 0x0100  | false           | 0
> 23  | BIOSStartupModuleEntry    (0x07) | 0xffc00000           | 262144   | 0x0100  | false           | 0
> 24  | BIOSPolicyRecord          (0x09) | 0x0                  | 0        | 0x0100  | false           | 0
> 25  | TXTPolicyRecord           (0x0A) | 0x2a050100710070     | 0        | 0x0000  | false           | 0
> 26  | KeyManifestRecord         (0x0B) | 0xfff0bd80           | 597      | 0x0100  | false           | 0
> 27  | BootPolicyManifestRecord  (0x0C) | 0xfff0b800           | 941      | 0x0100  | false           | 0
```

### CODE REVIEW

File: pkg/pcr/flow_detection.go [hunk 1]
```diff
@@ -131,7 +132,7 @@ func IsAMDPSPFirmware(firmware Firmware) bool {
 // no validation errors occurred.
 func DetectMainAttestationFlow(firmware Firmware, regs registers.Registers, tpmDevice tpmdetection.Type) (Flow, error) {
 	if IsAMDPSPFirmware(firmware) {
-		// TODO: whether TPM is initialised in locality 0 or 3 depends on the APCB binary tokens
```
Comment (trynity) [inline pkg/pcr/flow_detection.go:134]: American versus Hiberno/British, fight

File: go.mod [hunk 2]
```diff
@@ -18,14 +18,15 @@ require (
 	github.com/google/uuid v1.2.0
 	github.com/klauspost/cpuid/v2 v2.0.9
 	github.com/kr/pretty v0.2.1 // indirect
-	github.com/linuxboot/fiano v1.0.4-0.20211220133044-7de4aa4d8e3d
+	github.com/linuxboot/fiano v1.0.4-0.20220110153808-38194836b6f1
```
Comment (trynity) [inline go.mod:21]: I'm now worried about the fact that we're not on a major version yet for `fiano` and how future updates may make upgrading this/maintaining it painful. 

That's future me's problem I guess, but something to keep in mind

File: go.mod [hunk 3]
```diff
@@ -18,14 +18,15 @@ require (
...
-	github.com/stretchr/testify v1.6.1
+	github.com/stretchr/testify v1.7.0
 	github.com/tidwall/pretty v1.0.2
-	github.com/tjfoc/gmsm v1.4.0
+	github.com/tjfoc/gmsm v1.4.1
 	github.com/ulikunitz/xz v0.5.10
+	github.com/xaionaro-go/bytesextra v0.0.0-20220103144954-846e454ddea9 // indirect
 	github.com/xaionaro-go/gosrc v0.0.0-20201124181305-3fdf8476a735
-	github.com/xaionaro-go/unsafetools v0.0.0-20200202162159-021b112c4d30
+	github.com/xaionaro-go/unsafetools v0.0.0-20210722164218-75ba48cf7b3c
```
Comment (trynity) [inline go.mod:30]: Could you release major versions of these packages? Seems like it'd be good practice that we could follow.

**Review by trynity — APPROVED** [review]
Seems legit, mostly relying on CI to ensure no breakage.

File: pkg/pcr/flow_detection.go (hunk 1 above)
Comment (xaionaro) [reply pkg/pcr/flow_detection.go:134]: I just use spell-checker :)

File: go.mod (hunk 2 above)
Comment (xaionaro) [reply go.mod:21]: Sorry, cannot focus my mind. Could you rephrase, please?

Comment (trynity) [reply go.mod:21]: If we release a breaking API change in `fiano` for the parts that `converged-security-suite` depends on, unless we upgrade the major version of `fiano`, resolving dependencies will not be fun.

Comment (xaionaro) [reply go.mod:21]: No, I initially created these packages in fiano with updated API (to be more consistent with other fiano modules in terms of how they are used). There was no breaking change inside of fiano. The breaking change comes from switching itself (from CSS's version of the packages to fiano's version of the packages).

**Review by ChriMarMe — APPROVED** [review]
